### PR TITLE
CORCI-729 build: Add psm2 to libfabric build

### PR DESCRIPTION
**Pull request by brianjmurrell**
To match our current standards install psm2 into lib/ not lib64/.

Signed-off-By: Brian J. Murrell <brian.murrell@intel.com>

### CODE REVIEW

**Comment by brianjmurrell** [comment]
Failure in Github status is the "branch" build that typically fails for us.  The PR build succeeded.

**Review by frostedcmos — DISMISSED** [review]

File: components/__init__.py [hunk 1]
```diff
@@ -357,4 +363,15 @@ def define_components(reqs):
...
 
+    reqs.define('psm2',
+                retriever=GitRepoRetriever(
+                    'https://github.com/intel/opa-psm2.git'),
+                # psm2 hard-codes installing into /usr/...
+                commands=['sed -i -e "s/\\(.{DESTDIR}\\/\\)usr\\//\\1/" ' +
+                          '       -e "s/\\(INSTALL_LIB_TARG=\\/usr\\/lib\\)64/\\1/" ' +
+                          '       -e "s/\\(INSTALL_LIB_TARG=\\)\\/usr/\\1/" ' +
+                          'Makefile compat/Makefile',
+                          'make $JOBS_OPT', 'make DESTDIR=$PSM2_PREFIX install'],
```
[the remark this answers is not in the document]
Comment (jolivier23) [reply components/__init__.py]: this one seems simple enough...just put the second part on next line?

File: components/__init__.py [hunk 2]
```diff
@@ -357,4 +363,15 @@ def define_components(reqs):
                 libs=['protobuf-c'],
                 headers=['protobuf-c/protobuf-c.h'])
 
+    reqs.define('psm2',
+                retriever=GitRepoRetriever(
+                    'https://github.com/intel/opa-psm2.git'),
+                # psm2 hard-codes installing into /usr/...
+                commands=['sed -i -e "s/\\(.{DESTDIR}\\/\\)usr\\//\\1/" ' +
+                          '       -e "s/\\(INSTALL_LIB_TARG=\\/usr\\/lib\\)64/\\1/" ' +
```
[the remark this answers is not in the document]
Comment (jolivier23) [reply components/__init__.py]: add a pylint ignore?

**Review by jolivier23 — DISMISSED** [review]
changes look fine.  It would be good to keep the python linting clean

**Review by jolivier23 — APPROVED** [review]

**Review by frostedcmos — APPROVED** [review]

**Pull request merged**